packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -374,27 +374,26 @@ - (void)disableContextMenu
 - (void)handleLongPress:(UILongPressGestureRecognizer *)gesture
 {
 #if !TARGET_OS_UIKITFORMAC
-  if (@available(iOS 16.0, *)) {
-    CGPoint location = [gesture locationInView:self];
-    UIEditMenuConfiguration *config = [UIEditMenuConfiguration configurationWithIdentifier:nil sourcePoint:location];
-    if (_editMenuInteraction) {
-      [_editMenuInteraction presentEditMenuWithConfiguration:config];
-    }
-    return;
-  }
-  // TODO: Adopt showMenuFromRect (necessary for UIKitForMac)
-  UIMenuController *menuController = [UIMenuController sharedMenuController];
-
-  if (menuController.isMenuVisible) {
-    return;
-  }
-
-  if (!self.isFirstResponder) {
-    [self becomeFirstResponder];
+  if (@available(iOS 16.0, macCatalyst 16.0, *)) { // [visionOS]
```

Comment:
Is this marker necessary? This block looks in sync with upstream.

Reply:
Right. I think my change there but the time I made this PR ready for review.

Reply:
I cherry-picked https://github.com/facebook/react-native/commit/1dd647484aff00a079923a5c0e763e3f54ccc210 and https://github.com/facebook/react-native/commit/6801fc3f9bf172f5ca929127576558be04e3f99c into this PR as earlier commits. Therefore the last commit "Add native changes for visionOS" should be cleaner